src/jit/flowgraph.cpp

```diff
@@ -9315,8 +9315,14 @@ VARSET_VALRET_TP Compiler::fgGetVarBits(GenTreePtr tree)
     {
         VarSetOps::AddElemD(this, varBits, varDsc->lvVarIndex);
     }
-    else if (varDsc->lvType == TYP_STRUCT && varDsc->lvPromoted)
+    // We have to check type of root tree, not Local Var descriptor because
+    // for legacy backend we promote TYP_STRUCT to INT_INT in case unsed non-argument local.
```

Comment:
Should be "we promote TYP_STRUCT to TYP_INT if it is an unused or independently promoted non-argument struct local"